src/hu/openig/screen/items/PlanetScreen.java

```diff
@@ -596,8 +596,7 @@ public void invoke() {
 			}
 		});
 		if (surface() != null) {
-			render.offsetX = -(int)((surface().boundingRectangle.width * render.scale - width) / 2);
-			render.offsetY = -(int)((surface().boundingRectangle.height * render.scale - height) / 2);
+			centerScreen();
```

Comment:
The solution was not about centering the screen when changing planets but to position the camera relatively to the same location on the new planet considering the surface area of the new one. I.e., if I'm looking at the top-left position on a planet of 33x33 and switch to a 20x20 planet, I should be looking relatively at the same location in the center of the screen.

Reply:
I had a bit bigger change for storing the last view position (and zoom level), but i thought to first check in the smallest code change possible that addresses the issue as the first try.